catalog/admin/module/catalog-admin-module-maplayers/src/main/webapp/map-layers/reducer.js

```diff
@@ -45,7 +47,20 @@ export const fetch = () => (dispatch) => {
     .then((res) => res.json())
     .then((json) => {
       const config = fromJS(json).getIn(configPath)
-        .update('imageryProviders', (providers) => fromJS(JSON.parse(providers || '[]')))
+        .update('imageryProviders', (providers) => {
+          try {
+            const parsed = JSON.parse(providers)
+            const err = validateStructure(parsed)
+            if (err !== undefined) {
+              throw Error(err)
```

Comment:
I think if we hit this code path, things have gone seriously wrong, which is kind of exceptional. The whole PR is dealing with an edge case where the backend returns malformed map-layers configuration, which it should really never do, but just in case, the user should be allowed fix/discard the configuration.